### PR TITLE
Mastodon Layer

### PR DESCRIPTION
**Pull request by ksonney**
This adds a layer for connecting and posting to a mastodon social network instance. This layer uses mastodon.el from https://github.com/jdenen/mastodon.el. 

More information on mastodon.social can be found at https://mastodon.social

### CODE REVIEW

**Comment by JAremko** [comment]
@ksonney https://travis-ci.org/syl20bnr/spacemacs/jobs/224827643 trailing white spaces

**Comment by jdenen** [comment]
@ksonney The `mastodon-register` function is going away. It's old and unnecessary now. There's no need to include it in the layer.

**Comment by JAremko** [comment]
Still a couple of org file formatting errors :roll_eyes: 

Ignore errors in **.el* files.

**Comment by ksonney** [comment]
My git-fu isn't strong enough to remove the screwup where habitica was left in and had to be removed from this branch. That will likely cause issues with https://github.com/syl20bnr/spacemacs/pull/8682

**Comment by zestyr** [comment]
Try doing an interactive rebase where you drop those commits.

**Comment by ksonney** [comment]
OK, git rebase complete, caught up to current develop branch, and good to go.

File: layers/+web-services/mastodon/README.org [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#+TITLE: mastonon.social layer
```
Comment (Linuus) [inline layers/+web-services/mastodon/README.org]: mastodon? Not mastonon

**Comment by ksonney** [comment]
And with the MELPA package, this might actually be done-done (for now)

File: layers/+web-services/mastodon/packages.el [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+;;; License: GPLv3
+
+(defconst mastodon-packages
+  '(
+    mastodon))
+
+(defun mastodon/init-mastodon ()
+  ;; (use-package mastodon)
+  (require 'mastodon)
+  (setq mastodon-token-file "~/emacs.d/private/mastodon.plstore"))
```
Comment (syl20bnr) [inline layers/+web-services/mastodon/packages.el]: Why not using `use-package` ?

Comment (ksonney) [inline layers/+web-services/mastodon/packages.el]: Leftover from the pre-MELPA version. Updated and fix in the latest push.

**Pull request closed without merging**